src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -182,12 +197,29 @@ public async Task<bool> TryRefreshAsync()
             {
                 await RefreshAsync().ConfigureAwait(false);
             }
-            catch (Exception e) when (
-                e is KeyVaultReferenceException ||
-                e is RequestFailedException ||
-                ((e as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
-                e is OperationCanceledException)
+            catch (Exception ex) when (

```

Comment:
I'd like to keep the change you had where we aren't catching base `Exception`. It's generally a bad practice to catch base exception and do anything with it, even a `when` check. For example stackoverflowexception would be valid here, but Ideally we'd like that to bubble up asap.

The `when` clause is good if the type of the exception alone is not good enough to determine if you want to enter the catch block, but I'd consider using when to affirm the type a bit of an abuse.

Reply:
In this case it will result in two identical catch blocks, but there's not much of a choice with how ConfigurationClient propagates errors.

Reply:
Updated.